maestro/kubecontrol/monitor.go

```diff
@@ -0,0 +1,234 @@
+package kubecontrol
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-redis/redis/v8"
+	maestroconfig "github.com/ns1labs/orb/maestro/config"
+	"github.com/ns1labs/orb/pkg/errors"
+	sinkspb "github.com/ns1labs/orb/sinks/pb"
+	"go.uber.org/zap"
+	"io"
+	k8scorev1 "k8s.io/api/core/v1"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"strings"
+	"time"
+)
+
+const MonitorFixedDuration = 1 * time.Minute
+const TimeDiffActiveIdle = 5 * time.Minute
+
+func NewMonitorService(logger *zap.Logger, sinksClient *sinkspb.SinkServiceClient, redisClient *redis.Client, kubecontrol *Service) MonitorService {
+	return &monitorService{
+		logger:      logger,
+		sinksClient: *sinksClient,
+		redisClient: redisClient,
+		kubecontrol: *kubecontrol,
+	}
+}
+
+type MonitorService interface {
+	Start(ctx context.Context, cancelFunc context.CancelFunc) error
+}
+
+type monitorService struct {
+	logger      *zap.Logger
+	sinksClient sinkspb.SinkServiceClient
+	redisClient *redis.Client
+	kubecontrol Service
+}
+
+func (svc *monitorService) Start(ctx context.Context, cancelFunc context.CancelFunc) error {
+	go func(ctx context.Context, cancelFunc context.CancelFunc) {
+		ticker := time.NewTicker(MonitorFixedDuration)
+		svc.logger.Info("start monitor routine", zap.Any("routine", ctx))
+		defer func() {
+			cancelFunc()
+			svc.logger.Info("stopping monitor routine")
+		}()
+		for {
+			select {
+			case <-ctx.Done():
+				cancelFunc()
+				return
+			case _ = <-ticker.C:
+				svc.logger.Info("monitoring sinks")
+				svc.monitorSinks(ctx)
+			}
+		}
+	}(ctx, cancelFunc)
+	return nil
+}
+
+func (svc *monitorService) getPodLogs(ctx context.Context, pod k8scorev1.Pod) ([]string, error) {
+	maxTailLines := int64(10)
```

Comment:
Should probably pull this out into a constant

Reply:
I guess, so, but the scope is only for this func.